Lib/shutil.py

```diff
@@ -1415,7 +1415,7 @@ def which(cmd, mode=os.F_OK | os.X_OK, path=None):
             path.insert(0, curdir)
 
         # PATHEXT is necessary to check on Windows.
-        pathext = os.environ.get("PATHEXT", "").split(os.pathsep)
+        pathext = [ext for ext in os.environ.get("PATHEXT", "").split(os.pathsep) if ext]
```

Comment:
You could also fix the default value when `PATHEXT` isn't defined. It should be `".COM;.EXE;.BAT;.CMD;.VBS;.JS;.WS;.MSC"`. This is the default value that the CMD shell uses, as shown in a debugger:

```
0:000> du cmd!PathExtDefaultStr
00007ff6`ca393ff0  ".COM;.EXE;.BAT;.CMD;.VBS;.JS;.WS"
00007ff6`ca394030  ";.MSC"
```

Reply:
Fixed, added `_WIN_DEFAULT_PATHEXT` constant which will be used as default.